test/passing/break_cases_nested.ml

```diff
@@ -2,8 +2,7 @@
 
 let f x = function
   | C | P (this, test, [is; wide; enough; _to; break], [the; line]) | A | K
-    ->
-      1
+    -> 1
```

Comment:
With the 'nested' option, these breaks need to be preserved. The reasoning behind this is that `break-cases=nested` is intended for use with `indicate-nested-or-patterns=false`, and the breaks are needed to keep from hiding a short body within a sequence of cases that otherwise looks like a nested or-pattern.

Some people will prefer to take that risk, but we need to keep a the safe alternative.

Reply:
I chose to have a dedicated function for each case, I feel it's easier to maintain than having a generalized function for the 3 cases.

Reply:
I have to look more later, but initially I'm not sure about the split into separate functions. I agree that each one is simpler than the combined one, but there is still quite a bit of duplication of code that handles other things than just break-cases that we will need to keep in sync across the 3 functions.

Reply:
I factorized the code in a way it still shows where the box structure is different from one case to another.